test/utils.test.ts

```diff
@@ -0,0 +1,35 @@
+import { field, FieldDefinitions, validate, ValidationRule } from '../src'
+import { aWidget, Widget } from './utils'
+
+const pictureRequiresDescription: ValidationRule<string, Widget> = (
+  value,
+  state
+) => {
+  if (value && !state.details.description) {
+    return 'To upload a photo, you need to add a description'
+  }
+  return undefined
+}
+
+const fieldDefs: FieldDefinitions<Widget> = {
+  name: field(),
+  components: field(),
+  details: {
+    description: field(),
+    picture: field({ rules: [pictureRequiresDescription] })
+  }
+}
+
+describe('Helper function', () => {
+  it('should helper function to validate form data base on FieldDefinitions', () => {
+    expect(
+      validate(
+        aWidget({ details: { description: null as any, picture: 'bytes' } }),
+        fieldDefs
+      )
+    ).toBeFalsy()
+    expect(validate(aWidget({ name: '' }), fieldDefs)).toBeFalsy()
+
+    expect(validate(aWidget(), fieldDefs)).toBeTruthy()
```

Comment:
These two test cases both make sense.

In the spirit of having each test make assertions on a single scenario, could you please put this second scenario in its own test? I've found that keeping each scenario in its own test tends to make refactoring easier and, for longer tests, is easier to read.